botocore/loaders.py

```diff
@@ -103,21 +103,33 @@
 """
 import logging
 import os
+import pathlib
+import posixpath
+from zipfile import is_zipfile
+
+try:
+    from zipfile import Path as ZipPath
+except ImportError:
+
+    class ZipPath:
+        def __init__(self, path):
+            raise RuntimeError(
+                "Using botocore in a zipped context "
+                "is not supported in python < 3.9"
+            )
```

Comment:
Lets tie this directly to `HAS_ZIP_SUPPORT`. If we need to validate `zipfile` has Path available (I'm assuming all versions support?), move that into the initialization requirements for `HAS_ZIP_SUPPORT`.

Reply:
no `zipfile.Path` is only native as of python 3.8. regardless, because we try to import `importlib.resources.files` which is only a part of 3.9+, it will have the desired effect